---
layout: post
category: cstheoryrss
title: "arXiv: Computational Geometry: TrajLens: Visual Analysis for Constructing Cell Developmental"
date: 2025-07-22T00:00:00
---

**Authors:** [Qipeng Wang](https://dblp.uni-trier.de/search?q=Qipeng+Wang), [Shaolun Ruan](https://dblp.uni-trier.de/search?q=Shaolun+Ruan), [Rui Sheng](https://dblp.uni-trier.de/search?q=Rui+Sheng), [Yong Wang](https://dblp.uni-trier.de/search?q=Yong+Wang), [Min Zhu](https://dblp.uni-trier.de/search?q=Min+Zhu), [Huamin Qu](https://dblp.uni-trier.de/search?q=Huamin+Qu)

Constructing cell developmental trajectories is a critical task in
single-cell RNA sequencing (scRNA-seq) analysis, enabling the inference of
potential cellular progression paths. However, current automated methods are
limited to establishing cell developmental trajectories within individual
samples, necessitating biologists to manually link cells across samples to
construct complete cross-sample evolutionary trajectories that consider
cellular spatial dynamics. This process demands substantial human effort due to
the complex spatial correspondence between each pair of samples. To address
this challenge, we first proposed a GNN-based model to predict cross-sample
cell developmental trajectories. We then developed TrajLens, a visual analytics
system that supports biologists in exploring and refining the cell
developmental trajectories based on predicted links. Specifically, we designed
the visualization that integrates features on cell distribution and
developmental direction across multiple samples, providing an overview of the
spatial evolutionary patterns of cell populations along trajectories.
Additionally, we included contour maps superimposed on the original cell
distribution data, enabling biologists to explore them intuitively. To
demonstrate our system's performance, we conducted quantitative evaluations of
our model with two case studies and expert interviews to validate its
usefulness and effectiveness.

[Read original post](http://arxiv.org/abs/2507.15620v1)
